discovery/gossiper.go

```diff
@@ -361,23 +359,29 @@ type updatedChanPolicies struct {
 	err          error
 }
 
-// PropagateChanPolicyUpdate signals the AuthenticatedGossiper to update the
-// channel forwarding policies for the specified channels. If no channels are
-// specified, then the update will be applied to all outgoing channels from the
-// source node. Policy updates are done in two stages: first, the
+// EdgeWithInfo contains the information that is required to update an edge.
+type EdgeWithInfo struct {
+	// Info describes the channel.
+	Info *channeldb.ChannelEdgeInfo
+
+	// Edge describes the policy in one direction of the channel.
+	Edge *channeldb.ChannelEdgePolicy
+}
+
+// PropagateChanPolicyUpdate signals the AuthenticatedGossiper to perform the
+// specified edge updates. Updates are done in two stages: first, the
 // AuthenticatedGossiper ensures the update has been committed by dependent
 // sub-systems, then it signs and broadcasts new updates to the network. A
 // mapping between outpoints and updated channel policies is returned, which is
 // used to update the forwarding policies of the underlying links.
 func (d *AuthenticatedGossiper) PropagateChanPolicyUpdate(
-	newSchema routing.ChannelPolicy, chanPoints ...wire.OutPoint) (
+	edgesToUpdate []EdgeWithInfo) (
```

Comment:
Is this return value still needed if we now pass in the channel policy to be broadcast? Whereas before we used this to update the switch with what was returned by the gossiper.

Reply:
Good point. The only reason to keep it, is to properly deal with partial updates in the gossiper. But if we validate all policy updates in `localchans.Manager`, validation errors shouldn't occur. Only when there is a bug, the link could get desynchronized. I think that is reasonable. Carried out the simplification.